tests/layer_formatting_tests.py

```diff
@@ -105,6 +117,206 @@ def test_process_table(self):
+        |1    |2    |3   |4    |5     |6      |7     |8      |9      |10      |
+
+        This is the original XML (with just the first row):
+
+        <GPOTABLE CDEF="7,7,5,5,5,5,6,6,5,5" COLS="10" OPTS="L2">
+          <BOXHD>
+            <CHED H="1">Quantity of explosives</CHED>
+            <CHED H="2">Pounds over</CHED>
+            <CHED H="2">Pounds not over</CHED>
+            <CHED H="1">Distances in feet</CHED>
+            <CHED H="2">Inhabited buildings</CHED>
+            <CHED H="3">Barri-caded</CHED>
+            <CHED H="3">Unbarri-caded</CHED>
+            <CHED H="2">Public highways with traffic volume of 3000 or fewer vehicles/day</CHED>
+            <CHED H="3">Barri-caded</CHED>
+            <CHED H="3">Unbarri-caded</CHED>
+            <CHED H="2">Passenger railways—public highways with traffic volume of more than 3,000 vehicles/day</CHED>
+            <CHED H="3">Barri-caded</CHED>
+            <CHED H="3">Unbarri-caded</CHED>
+            <CHED H="2">Separation of magazines</CHED>
+            <CHED H="3">Barri-caded</CHED>
+            <CHED H="3">Unbarri-caded</CHED>
+          </BOXHD>
+          <ROW>
+            <ENT I="01">0</ENT>
+            <ENT>5</ENT>
+            <ENT>70</ENT>
+            <ENT>140</ENT>
+            <ENT>30</ENT>
+            <ENT>60</ENT>
+            <ENT>51</ENT>
+            <ENT>102</ENT>
+            <ENT>6</ENT>
+            <ENT>12</ENT>
+          </ROW>
+        </GPOTABLE>
+
+        """
+        with self.tree.builder("GPOTABLE", CDEF="7,7,5,5,5,5,6,6,5,5",
+                               COLS="10", OPTS="L2") as root:
+            with root.BOXHD() as hd:
+                hd.CHED(u"Quantity of explosives", H=1)
+                hd.CHED(u"Pounds over", H=2)
+                hd.CHED(u"Pounds not over", H=2)
+                hd.CHED(u"Distances in feet", H=1)
+                hd.CHED(u"Inhabited buildings", H=2)
+                hd.CHED(u"Barri-caded", H=3)
+                hd.CHED(u"Unbarri-caded", H=3)
+                hd.CHED(u"Public highways with traffic volume of 3000 or fewer vehicles/day", H=2)
+                hd.CHED(u"Barri-caded", H=3)
+                hd.CHED(u"Unbarri-caded", H=3)
+                hd.CHED(u"Passenger railways—public highways with traffic volume of more than 3,000 vehicles/day", H=2)
+                hd.CHED(u"Barri-caded", H=3)
+                hd.CHED(u"Unbarri-caded", H=3)
+                hd.CHED(u"Separation of magazines", H=2)
+                hd.CHED(u"Barri-caded", H=3)
+                hd.CHED(u"Unbarri-caded", H=3)
+            with root.ROW() as row:
+                row.ENT(u"0", I="01")
+                row.ENT(u"5")
+                row.ENT(u"70")
+                row.ENT(u"140")
+                row.ENT(u"30")
+                row.ENT(u"60")
+                row.ENT(u"51")
+                row.ENT(u"102")
+                row.ENT(u"6")
+                row.ENT(u"12")
```

Comment:
This regression test is nice to have. Is there a way to reduce the repetition, though? If you want to include the full XML, you don't also need to use the XML builder -- just use `etree.fromstring`